ArchiSteamFarm/Bot.cs

```diff
@@ -2865,40 +2867,61 @@ public sealed class Bot : IAsyncDisposable {
 				if (string.IsNullOrEmpty(lastPage)) {
 					ArchiLogger.LogNullError(nameof(lastPage));
 
-					return new HashSet<uint>(0);
+					return null;
 				}
 
 				if (!byte.TryParse(lastPage, out maxPages) || (maxPages == 0)) {
 					ArchiLogger.LogNullError(nameof(maxPages));
 
-					return new HashSet<uint>(0);
+					return null;
 				}
 			}
 
-			HashSet<uint> appIDs = GetPossiblyCompletedBadgeAppIDs(badgePage);
-			byte currentPage = 1;
+			HashSet<uint>? firstPageIDs = GetPossiblyCompletedBadgeAppIDs(badgePage);
+
+			if (firstPageIDs == null) {
+				return null;
+			}
 
 			// We need to check all badge pages, as a badge which we already crafted to max level might hide somewhere in the back
-			while (currentPage <= maxPages) {
-				appIDs.UnionWith(await GetPossiblyCompletedBadgeAppIDs(++currentPage).ConfigureAwait(false));
+			HashSet<Task<HashSet<uint>?>> tasks = new HashSet<Task<HashSet<uint>?>>(maxPages - 1);
+
+			for (byte page = 2; page < maxPages; ++page) {
+				// We need a copy of variable being passed when in for loops, as loop will proceed before our task is launched
+				byte currentPage = page;
+				tasks.Add(GetPossiblyCompletedBadgeAppIDs(currentPage));
```

Comment:
This will start the task immediately and therefore not comply with `MinMemoryUsage`.